src/VisualStudio/Core/Def/Implementation/TableDataSource/AbstractTableDataSource.cs

```diff
@@ -136,7 +136,9 @@ protected void OnDataAddedOrChanged(object data)
             lock (_gate)
             {
                 snapshot = _subscriptions;
+#pragma warning disable CA2000 // Dispose objects before losing scope - _map has the dispose ownership for the factory.
```

Comment:
❓ I don't see how `_map` has ownership here. Can you describe the sequence that leads to resource cleanup?

Reply:
I will file a tracking bug to investigate this instance.